### PR TITLE
B/database

### PR DESCRIPTION
**Pull request by worldofjoni**
# Pull Request for initial Implementation

## Backend
Make sure, you have done de following before merging this pull request:
- [x] Your code is **documented** using doc comments.
- [x] You added appropriate **logging** in your code.
- [x] You added **unit tests** to all your implemented functions.
- [x] You ran `cargo format` to format your code.
- [x] You ran `cargo clippy` to check for improvements.
- [x] You added the **class descriptions** of the classes you implemented in the **latex document** on overleaf.
      Make sure to mark any changes you made to the design documents specification.

### CODE REVIEW

File: backend/migrations/20230703131255_food_plan.up.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+-- Add up migration script here
+
+CREATE DOMAIN price AS integer CHECK (VALUE >= 0);
+
+CREATE TABLE food_plan (
+  line_id uuid NOT NULL,
+  food_id uuid NOT NULL,
+  date date NOT NULL,
```
Comment (Whatsuup) [inline backend/migrations/20230703131255_food_plan.up.sql:8]: `date date` => no compile error?

File: backend/migrations/20230703132445_image_rating.up.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- Add up migration script here
+CREATE DOMAIN vote as smallint CHECK (VALUE = -1 OR VALUE = 1);
```
Comment (Whatsuup) [inline backend/migrations/20230703132445_image_rating.up.sql:2]: DEFAULT val = 0 for images without user vote?

Comment (worldofjoni) [reply backend/migrations/20230703132445_image_rating.up.sql:2]: no vote == no tuple in relation

File: backend/migrations/20230703131255_food_plan.up.sql (hunk 1 above)
Comment (worldofjoni) [reply backend/migrations/20230703131255_food_plan.up.sql:8]: it complied... but we can rename it to `serve_date`

**Comment by worldofjoni** [comment]
The Image-Rank function definition is still missing.

File: backend/migrations/20230703131255_food_plan.up.sql (hunk 1 above)
Comment (ElenaHaeussler) [reply backend/migrations/20230703131255_food_plan.up.sql]: Das sind Sprachenabhängige Implementierungsdetails

**Review by ElenaHaeussler — APPROVED** [review]

**Pull request merged**